flight/Modules/Actuator/actuator.c

```diff
@@ -413,37 +413,80 @@ float ProcessMixer(const int index, const float curve1, const float curve2,
 	return(result);
 }
 
-
 /**
- *Interpolate a throttle curve. Throttle input should be in the range 0 to 1.
- *Output is in the range 0 to 1.
+ * Interpolate a mixer curve
+ *
+ * output range is defined by the curve vector
+ * curve is defined in N intervals connecting N+1 points
+ *
+ * @param input the input value, in [input_min,input_max]
+ * @param curve the array of points in the curve
+ * @param num_points the number of points in the curve
+ * @param input_min the lower range of the input values
+ * @param input_max the upper range of the input values
+ * @return the output value, in [0,1]
  */
-static float MixerCurve(const float throttle, const float* curve, uint8_t elements)
+static float MixerCurve(float const input, float const * curve, uint8_t num_points, const float input_min, const float input_max)
 {
-	float scale = throttle * (float) (elements - 1);
+	// shift our input [min,max] into the typical range [0,1]
+	float scale = ( (input - input_min) / (input_max - input_min) ) * (float) (num_points - 1);
+	// select a starting bin via truncation
 	int idx1 = scale;
```

Comment:
@mlyle OK that makes sense. BTW I tested the `fmax` version and it passes the test vector. Let me know which you think is better and I'll commit the change.